src/julia.jl

```diff
@@ -33,4 +33,139 @@ function multireshape(parent::AbstractArray, dimss::Array{Tuple{Vararg{Int}}})
     multireshape(parent, dimss...)
 end
 
+
+"""
+    pairwise(f::Function, a; kwargs...)
+
+Apply the function `f` to all pairwise combinations of elements from `a`. Set
+keyword `symmetric = true` to only calculate `f(x,y)` when `f(x,y) == f(y,x)`.
+"""
+function pairwise(f::Function, a; symmetric = false)
+    n = length(a)
+    f = tryfunc(f, a[1], a[1])
+    first = f(a[1], a[1])
+    r = Matrix{typeof(first)}(n, n)
+    pairwise!(f, a, r; symmetric = symmetric)
+    r
+end
+
+"""
+    pairwise(f::Function, a, output; kwargs...)
+
+Apply the function `f` to all pairwise combinations of elements from `a`, and
+return a `Vector` of the results if `output = :Vector`. The Vector form is
```

Comment:
I respectfully disagree - since only `output = :Vector` is defined, if not specified it will return a Matrix. Thus, the type of the return value can be deduced by the compiler based only on the type of the arguments (in this case, whether the `output` argument is passed or not).